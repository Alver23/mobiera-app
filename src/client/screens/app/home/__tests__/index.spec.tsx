// Dependencies
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { View } from 'react-native';
import { Button } from 'react-native-elements';
import * as NavigationHooks from '@react-navigation/native';

// Under test file
import HomeScreen from '../index';

// Mocks
jest.mock('@react-navigation/native');

describe('<HomeScreen />', () => {
  let component: ShallowWrapper;

  const navigation: any = {
    navigate: jest.fn(),
  };

  beforeEach(() => {
    jest.spyOn(NavigationHooks, 'useNavigation').mockReturnValue(navigation);
    component = shallow(<HomeScreen />);
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render correctly', () => {
    expect(component.find(View).exists()).toBeTruthy();
  });

  it('should navigate to the screen edit user', () => {
    const onPress = component.find(Button).at(1).prop('onPress');
    onPress();

    expect(navigation.navigate).toHaveBeenCalledTimes(1);
  });
});
