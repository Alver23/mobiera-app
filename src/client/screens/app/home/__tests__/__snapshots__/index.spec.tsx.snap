// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> should save a snapshot of the component 1`] = `
<ScrollView
  contentContainerStyle={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 2,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <UserProfile />
  </View>
  <Themed.Button
    title="Logout"
  />
</ScrollView>
`;
